src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,205 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (head == null) {
+            head = node;
+            tail = head;
+        } else {
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isValidIndexForAdd(index);
+        if (isEmpty()) {
+            add(value);
+        } else if (index == 0) {
+            addToHead(value);
+        } else {
+            Node<T> previous = findNodeByIndex(index);
+            if (previous == null) {
+                addToTail(value);
+            } else {
+                insert(value, previous);
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isValidIndexForSetAndGet(index);
+        return findNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isValidIndexForSetAndGet(index);
+        Node<T> foundNode = findNodeByIndex(index);
+        T oldValue = foundNode.value;
+        foundNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> deletedElement = findNodeByIndex(index);
+        if (deletedElement == null) {
+            throw new IndexOutOfBoundsException("wrond index: " + index + ", for empty list");
+        }
```

Comment:
redundant `if` clause, we already check index ant throws exception inside `isValid..()` method